### PR TITLE
Add python3-graphviz to rosdep/python.yaml

### PR DESCRIPTION
**Pull request by frankkusters**
Please add the following dependency to the rosdep database.

## Package name:

python3-graphviz

## Package Upstream Source:

Source: https://github.com/xflr6/graphviz
Docs: https://graphviz.readthedocs.io/

## Purpose of using this:

This package is a pure Python implementation for creating dot graphs, which can then be rendered with Graphviz.

It was added to rosdep as a pip entry in pull request https://github.com/ros/rosdistro/pull/16412 . That was 2017, by now many Linux distros have this available as native packages.

There are other rules with similar names:
- `python-graphviz-pip`: this is the pip version
- `python-pygraphviz` and `python3-pygraphviz`: Python wrappers for the Graphviz C library
- `graphviz`: Graphviz itself.

The rule for `python-pygraphviz` had incorrect references for Arch (non-existent) and Alpine ([package link](https://pkgs.alpinelinux.org/packages?name=py3-pygraphviz&branch=edge&repo=&arch=&maintainer=)), which I also fixed.

Distro packaging links:

- Debian: https://packages.debian.org/search?keywords=python3-graphviz&searchon=names&suite=all&section=all
  - REQUIRED
- Ubuntu: https://packages.ubuntu.com/search?keywords=python3-graphviz&searchon=names&suite=all&section=all
   - REQUIRED
- Fedora: https://packages.fedoraproject.org/pkgs/python-graphviz/python3-graphviz/
  - IF AVAILABLE
- Arch: https://archlinux.org/packages/community/any/python-graphviz/
  - IF AVAILABLE
- Gentoo: https://packages.gentoo.org/packages/dev-python/graphviz
  - IF AVAILABLE
- macOS: not available
  - IF AVAILABLE
- Alpine: https://pkgs.alpinelinux.org/packages?name=py3-pygraphviz&branch=edge&repo=&arch=&maintainer=
  - IF AVAILABLE
- NixOS/nixpkgs: [link](https://search.nixos.org/packages?channel=22.05&show=python39Packages.graphviz&from=0&size=50&buckets=%7B%22package_attr_set%22%3A%5B%22python39Packages%22%5D%2C%22package_license_set%22%3A%5B%5D%2C%22package_maintainers_set%22%3A%5B%5D%2C%22package_platforms%22%3A%5B%5D%7D&sort=relevance&type=packages&query=graphviz)
  - OPTIONAL
- openSUSE: https://software.opensuse.org/package/python3-graphviz
  - IF AVAILABLE

### CODE REVIEW

File: rosdep/python.yaml
Comment (wjwwood) [inline rosdep/python.yaml:8337]: Makes sense, thanks for fixing this one up too.

**Review by wjwwood — APPROVED** [review]
lgtm, but it needs a second review.

Also, you might want to look into the warning from CI:

> Key 'python3-graphviz' might be satisifed by opensuse package named 'python3-graphviz': https://software.opensuse.org/package/python-graphviz

Maybe fill it out for opensuse.

**Comment by frankkusters** [comment]
> Also, you might want to look into the warning from CI:
> 
> > Key 'python3-graphviz' might be satisifed by opensuse package named 'python3-graphviz': https://software.opensuse.org/package/python-graphviz
> 
> Maybe fill it out for opensuse.

Thanks for the heads-up. I've added it.

I was confused why I hadn't found the package. Turns out searching for ["graphviz"](https://software.opensuse.org/search?baseproject=ALL&q=graphviz) at openSUSE yields completely different results than ["python graphviz"](https://software.opensuse.org/search?baseproject=ALL&q=python+graphviz), while I was expecting a superset. I'll make sure to do a full name search next time.

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**
